src/validations.cr

```diff
@@ -147,8 +147,20 @@ module Validations
         previous_def
       {% end %}
 
-      {% for rule in rules.keys %}
-        validate_{{rule.id.gsub(/\s/, "_")}}({{attribute.stringify}}, {{attribute}}, {{rules[rule]}})
+      {% if rules[:if] %}
+        if {{ rules[:if] }}.call(self)
```

Comment:
So, to dry up this specific part you can do it like this:

```crystal
{% if rules[:if] %}
  if {{rules[:if]}}.call(self)
{% end %}

{% if rules[:unless] %}
  unless {{rules[:unless]}}.call(self)
{% end %}

dried_stuff

{% if rules[:if] %}
  end
{% end %}

{% if rules[:unless] %}
  end
{% end %}
```

Reply:
That was fast :laughing:

Reply:
Oh gosh I've just pushed a refactoring 😅 

I think this comment is related to the previous "unrefactored" version.